rag/enhanced_rag.py

```
@@ -0,0 +1,158 @@
+        response = client.chat.completions.create(
+            model="gpt-4o-mini",
+            messages=[{"role": "user", "content": prompt}],
+            temperature=0.3
+        )
+
+        scores = []
+        for line in response.choices[0].message.content.split('\n'):
+            if ':' in line:
+                try:
+                    scores.append(float(line.split(':')[1].strip().split()[0]))
+                except:
```

Ruff (0.13.2)

Do not use bare `except`

(E722)